### PR TITLE
Have conditional_var_value_t return correct matrix type if scalar is var

### PR DESCRIPTION
**Pull request by SteveBronder**
## Summary

This just makes `conditional_var_value_t<var, Eigen::Matrix<var, -1, -1>>` returns back a `var_value<Eigen::Matrix<double, -1, -1>>`

## Tests

Test added for checking above behavior

```
./runTests.py ./test/unit/math/rev/meta/conditional_var_value_test.cpp
```

## Side Effects

Nope

## Release notes

Fixes case for `conditional_var_value_t`

## Checklist

- [x] Math issue #1805 

- [x] Copyright holder: (fill in copyright holder information)

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Review by t4c1 — CHANGES_REQUESTED** [review]
Fix the doxygen according to the change.

File: stan/math/rev/meta/conditional_var_value.hpp [hunk 1]
```diff
@@ -8,7 +8,9 @@
 namespace stan {
 
 /**
- * Constructs a prim type or var_value from a scalar and a container.
+ * Conditionally construct a var_value container based on a scalar type. For
+ * var types as the scalar, the `var_value<Matrix>`'s inner type will have a
+ *  scalar of a double.
  * @tparam T_scalar Determines the scalar (var/double) of the type.
  * @tparam T_container Determines the container (matrix/vector/matrix_cl ...) of
  * the type. This must be a prim type.
```
Comment (t4c1) [inline stan/math/rev/meta/conditional_var_value.hpp:16]:
```suggestion
 * the type.
```
I meant this part. Sorry for the confusion.

**Review by t4c1 — APPROVED** [review]

**Comment by SteveBronder** [comment]
@rok-cesnovar is it fine to merge this? It's a bug fix

**Comment by rok-cesnovar** [comment]
Yeah, bugfixes are good.

**Pull request merged**